--- ./source3/registry/reg_init_basic.cbak	2017-01-10 20:01:08.812791125 +0800
+++ ./source3/registry/reg_init_basic.c	2017-01-10 20:01:36.988067137 +0800
@@ -45,11 +45,13 @@
 	}
 
 	/* setup the necessary keys and values */
-
-	werr = init_registry_data();
-	if (!W_ERROR_IS_OK(werr)) {
-		DEBUG(0, ("Failed to initialize data in registry!\n"));
-	}
+        /* setup the necessary keys and values */
+        if (!regdb_is_readonly ()) {
+            werr = init_registry_data();
+            if (!W_ERROR_IS_OK(werr)) {
+                DEBUG(0, ("Failed to initialize data in registry!\n"));
+            }
+        }
 
 done:
 	return werr;
