--- ./source3/registry/reg_init_smbconf.c.bak	2017-01-10 20:02:24.024528619 +0800
+++ ./source3/registry/reg_init_smbconf.c	2017-01-10 20:04:30.037762542 +0800
@@ -51,12 +51,14 @@
 		goto done;
 	}
 
-	werr = init_registry_key(keyname);
-	if (!W_ERROR_IS_OK(werr)) {
-		DEBUG(1, ("Failed to initialize registry key '%s': %s\n",
-			  keyname, win_errstr(werr)));
-		goto done;
-	}
+        if (!regdb_is_readonly ()) {
+            werr = init_registry_key(keyname);
+            if (!W_ERROR_IS_OK(werr)) {
+                DEBUG(1, ("Failed to initialize registry key '%s': %s\n",
+                      keyname, win_errstr(werr)));
+                goto done;
+            }
+        }
 
 	werr = reghook_cache_add(keyname, &smbconf_reg_ops);
 	if (!W_ERROR_IS_OK(werr)) {
