--- ./wscript.bak	2016-12-16 10:39:15.296654556 +0800
+++ ./wscript	2016-12-16 10:39:59.813319469 +0800
@@ -162,7 +162,6 @@
     if not conf.CHECK_CODE('#include "tests/summary.c"',
                            define='SUMMARY_PASSES',
                            addmain=False,
-                           execute=True,
                            msg='Checking configure summary'):
         raise Utils.WafError('configure summary failed')
     
