--- ./buildtools/wafadmin/Tools/python.py.bak	2016-12-16 11:18:42.599894589 +0800
+++ ./buildtools/wafadmin/Tools/python.py	2016-12-16 11:20:54.969889372 +0800
@@ -192,6 +192,19 @@
 """ % (python, python_prefix, python_SO, python_SYSLIBS, python_LDFLAGS, python_SHLIBS,
 	python_LIBDIR, python_LIBPL, INCLUDEPY, Py_ENABLE_SHARED, python_MACOSX_DEPLOYMENT_TARGET))
 
+        # Allow some python overrides from env vars for cross-compiling
+        os_env = dict(os.environ)
+
+        override_python_LDFLAGS = os_env.get('python_LDFLAGS', None)
+        if override_python_LDFLAGS is not None:
+                conf.log.write("python_LDFLAGS override from environment = %r\n" % (override_python_LDFLAGS))
+                python_LDFLAGS = override_python_LDFLAGS
+
+        override_python_LIBDIR = os_env.get('python_LIBDIR', None)
+        if override_python_LIBDIR is not None:
+                conf.log.write("python_LIBDIR override from environment = %r\n" % (override_python_LIBDIR))
+                python_LIBDIR = override_python_LIBDIR
+
 	if python_MACOSX_DEPLOYMENT_TARGET:
 		conf.env['MACOSX_DEPLOYMENT_TARGET'] = python_MACOSX_DEPLOYMENT_TARGET
 		conf.environ['MACOSX_DEPLOYMENT_TARGET'] = python_MACOSX_DEPLOYMENT_TARGET
